src/oop/special_class_method/medium/q12/MainTest.java

```diff
@@ -0,0 +1,143 @@
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "Shirt";
+    private final String attributeName1 = "size";
+    private final String getAttributeMethodName1 = "getSize";
+    private final String attributeName2 = "yearMade";
+    private final String methodName = "displaySize";
+    private static final int yearPivot = 2000;
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q12." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + " for the %s attribute. Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName1, int.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName1));
+        assertTrue(testClass.hasField(attributeName2, int.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName1));
+        Object[][] tests = new Object[][]{
+                {3, yearPivot + 2},
+                {3, yearPivot + 1},
+                {10, yearPivot -50},
+                {12, yearPivot + 20},
+                {43, yearPivot + 3},
+                {7, yearPivot - 5},
+                {7, yearPivot}
+        };
+        modifiedClassMessage =
+                "You have modified the provided portions of class " + className + " for the constructor. Please revert them to the original state.";
+        for (int i = 0; i < tests.length; i++) {
+            int size = (int) tests[i][0];
+            int yearMade = (int) tests[i][1];
+            Object[][] arguments = {
+                    {size, int.class},
+                    {yearMade, int.class}
+            };
+            Class<?>[] classes = {int.class, int.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(size, testClass.getFieldValue(testInstance, attributeName1), modifiedClassMessage);
+            _assertEquals(yearMade, testClass.getFieldValue(testInstance, attributeName2), modifiedClassMessage);
+            assertTrue(testClass.hasModifier(classes, "public"), modifiedClassMessage);
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(3, yearPivot + 12),
+                Arguments.of(3, yearPivot + 1),
+                Arguments.of(3, yearPivot - 1),
+                Arguments.of(10, yearPivot - 6),
+                Arguments.of(12, yearPivot - 26),
+                Arguments.of(43, yearPivot + 23),
+                Arguments.of(7, yearPivot)
+        );
+    }
+
+    @Test
+    public void getSizeIsDefinedCorrectly() {
```

Comment:
I don't believe we need this style of test since all this gets tested in another test but check with Seth first